File: C/CWE-416.c
#include <stdio.h>
#include <stdlib.h>

void use_after_free() {
    int *ptr = (int *)malloc(sizeof(int) * 10);
    if (ptr == NULL) {
        printf("Ошибка выделения памяти\n");
        return;
    }
    
    free(ptr); // Освобождение памяти

    ptr[0] = 42; // Использование памяти после освобождения (Use After Free)
    printf("Первый элемент: %d\n", ptr[0]);
}

int main() {
    use_after_free();
    return 0;
}

File: GO/CWE-416.go
package main

import (
	"fmt"
)

func useAfterFree() {
	arr := make([]int, 10)
	arr = nil // Освобождение памяти

	arr[0] = 42 // Использование памяти после освобождения (Use After Free)
	fmt.Println(arr[0])
}

func main() {
	useAfterFree()
}
